src/av2/structures/sweep.py

```diff
@@ -89,3 +91,71 @@ def from_feather(cls, lidar_feather_path: Path) -> Sweep:
             ego_SE3_up_lidar=ego_SE3_up_lidar,
             ego_SE3_down_lidar=ego_SE3_down_lidar,
         )
+
+    def equalize_intensity_distribution(self) -> None:
+        """Re-distribute mass of distribution."""
+        self.intensity = equalize_distribution(self.intensity)
+
+    def transform_from(self, target_SE3_src: SE3) -> Sweep:
+        """Transform 3d points in the sweep to a new reference frame.
+
+        Note: This method exists so that `SE3` can play around with it, without a cyclic dependency.
+
+        Args:
+            target_SE3_src: SE(3) transformation. Assumes the sweep points are provided in the `src` frame.
+
+        Returns:
+            a new Sweep object, with points provided in the the `target` frame.
+        """
+        return Sweep(
+            xyz=target_SE3_src.transform_from(self.xyz),
+            intensity=self.intensity,
+            laser_number=self.laser_number,
+            offset_ns=self.offset_ns,
+            timestamp_ns=self.timestamp_ns,
+            ego_SE3_up_lidar=self.ego_SE3_up_lidar,
+            ego_SE3_down_lidar=self.ego_SE3_down_lidar,
+        )
+
+    def stack(self, sweep: Sweep) -> None:
+        """Stack sweeps. Used for lidar sweep aggregation.
+
+        Args:
+            sweep: Lidar sweep.
+        """
+        self.xyz = np.vstack([self.xyz, sweep.xyz])
+        self.intensity = np.concatenate([self.intensity, sweep.intensity])  # type: ignore
+        self.laser_number = np.concatenate([self.laser_number, sweep.laser_number])  # type: ignore
+
+
+def normalize_array(array: NDArrayFloat) -> NDArrayFloat:
+    """Normalize array values, i.e. bring them to a range of [0,1].
+
+    Args:
+        array: Numpy array of any shape, representing unnormalized values.
+
+    Returns:
+        Numpy array of any shape, representing normalized values.
+    """
+    array -= array.min()  # shift min val to 0
+    array /= array.max()  # shrink max val to 1
+    return array
+
+
+def equalize_distribution(reflectance: NDArrayByte) -> NDArrayByte:
```

Comment:
```suggestion
def equalize_distribution(intensity: NDArrayByte) -> NDArrayByte:
```

Can we call this `intensity` to be consistent with naming?

Reply:
Sounds good to me.